modules/programs/fish.nix

```diff
@@ -151,12 +137,7 @@ in
       status --is-interactive; and not set -q __fish_nix_darwin_interactive_config_sourced
       and begin
         ${fishAliases}
-        
 
-        set fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions $fish_function_path
-        fenv source /etc/fish/foreign-env/interactiveShellInit > /dev/null
-        set -e fish_function_path[1]
-        
```

Comment:
You're removing a bunch of the shell inits here, which will break gpg-agent for example.

I think we should translate all of the init files and save them as `environment.etc."fish/interactiveShellInit.fish"` etc. Then we also don't need `system.build.setEnvironmentFish`

Reply:
@bouk I've tried implementing this approach. If you could have a look and give feedback, I'll port the same change to the nixOS module.